snow/engine/common/tracker/peers.go

```diff
@@ -272,14 +271,21 @@ func (p *peerData) SampleValidator() (ids.NodeID, bool) {
 	return p.connectedValidators.Peek()
 }
 
-func (p *peerData) GetValidators() set.Set[ids.NodeID] {
-	return set.Of(maps.Keys(p.validators)...)
+func (p *peerData) GetValidators() set.Set[ids.NodeWeight] {
+	res := set.NewSet[ids.NodeWeight](len(p.validators))
+	for k, v := range p.validators {
+		res.Add(ids.NodeWeight{Node: k, Weight: v})
+	}
+	return res
 }
 
-func (p *peerData) ConnectedValidators() set.Set[ids.NodeID] {
+func (p *peerData) ConnectedValidators() set.Set[ids.NodeWeight] {
```

Comment:
not sure if this would be racy though

Reply:
Even if it won't be racy, we don't want to do that, because it would introduce a ton of lock contention.
Grabbing a lock once and copying the data, is more efficient than grabbing the lock each time to get a tiny piece of data, many many times.